### DISCUSSION TITLE
rule  '@typescript-eslint/no-non-null-assertion' is not working

### DISCUSSION THREAD

**Discussion by jrubner** [angular-eslint/angular-eslint#1883] [Q&A] [answered]
`rules: {
      '@typescript-eslint/no-non-null-assertion': 'error',
      ...
`

i printed the active rules to a config-output:

`"@typescript-eslint/no-non-null-assertion": [
      2
    ],
`

So the rule should be active, right? Am i missing something?

` hello!: string; //TODO: this should not work! Linter should throw an error`

But it just says **"All files pass linting."**

Other rules like: `@typescript-eslint/explicit-function-return-type` work with my config.

Also: is there like a list of rules or an example of the new config file? I am new to this and i only find discussions for the old config file.

**Comment by JamesHenry** [comment, marked as answer]
That is not a non-null assertion, it is a definite assignment assertion, a separate language feature that uses the same character `!` as its operator: https://github.com/typescript-eslint/typescript-eslint/issues/4247

The typescript-eslint configs are transparently referenced from your config, angular-eslint does not reference them behind the scenes so I think what you want is https://typescript-eslint.io/users/configs

**Reply by jrubner** [reply]
Thanks for the clarification, very helpful!
